### ISSUE TITLE
git-state produces unpredicted results on Windows machine

### ISSUE THREAD

**Issue by pankajpatel** [watson/git-state#4]
Hi,

I was trying to use this repo in my cli utility [list-repos](https://github.com/pankajpatel/list-repos).

Recently I updated my utility from `simple-git` to `git-state` for the normal git statuses around repo, I used the new updated code on Windows and it failed.

Unpredictable results included empty branch-names in many cases.

Then I tried your similar repo `code-status`, which worked fine on Mac but that also failed on Windows. I think git-state has few things that should be optimized for Windows.

**Comment by pankajpatel** [comment]
following is the output of test.js on windows

```
Z:\git-state>node test.js
TAP version 13
# #isGit()
ok 1 should be equal
# #isGitSync()
ok 2 should be equal
# #check()
ok 3 null
ok 4 should be equivalent
not ok 5 should be equal
  ---
    operator: equal
    expected: 'string'
    actual:   'undefined'
    at: _combinedTickCallback (internal/process/next_tick.js:67:7)
  ...
ok 6 should be equal
ok 7 should be equal
ok 8 should be equal
ok 9 should be equal
# #untracked()
ok 10 null
ok 11 should be equal
# #dirty()
ok 12 null
ok 13 should be equal
# #branch()
ok 14 undefined
not ok 15 should be equal
  ---
    operator: equal
    expected: 'string'
    actual:   'undefined'
    at: maybeClose (internal/child_process.js:877:16)
  ...
# #ahead()
ok 16 null
ok 17 should be equal
# #commit()
ok 18 null
ok 19 should be equal

1..19
# tests 19
# pass  17
# fail  2
```

**Comment by watson** [comment]
I've published your pull request that fixes this bug to npm as v2.5.2
